framework/framework.go

```diff
@@ -164,6 +170,16 @@ func (f *Framework) UpdateFunc(oldObj, newObj interface{}) {
 	f.mutex.Lock()
 	defer f.mutex.Unlock()
 
+	ok, err := f.addFinalizer(obj)
+	if err != nil {
+		f.logger.Log("event", "update", "function", "UpdateFunc", "level", "error", "message", "stop framework reconciliation due to error", "stack", fmt.Sprintf("%#v", err))
+		return
+	}
+	if ok {
+		f.logger.Log("event", "update", "function", "UpdateFunc", "level", "debug", "message", "stop framework reconciliation, finalizer added")
```

Comment:
Not sure if this debug message is clear enough or helpful.

Reply:
I would like to align the wording with the other messages. It should then read `due to finalizer added`. It would be good to have code comments why we stop reconciliation here because this is not really obvious.

Reply:
+1 That's why added my comment below. `due to` could save us from that.